src/Symfony/Component/Notifier/Bridge/SmsBiuras/SmsBiurasTransportFactory.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\SmsBiuras;
+
+use Symfony\Component\Notifier\Exception\UnsupportedSchemeException;
+use Symfony\Component\Notifier\Transport\AbstractTransportFactory;
+use Symfony\Component\Notifier\Transport\Dsn;
+use Symfony\Component\Notifier\Transport\TransportInterface;
+
+/**
+ * @author Vasilij Duško <vasilij@prado.lt>
+ */
+final class SmsBiurasTransportFactory extends AbstractTransportFactory
+{
+    /**
+     * @return SmsBiurasTransport
+     */
+    public function create(Dsn $dsn): TransportInterface
+    {
+        $scheme = $dsn->getScheme();
+
+        if ('smsbiuras' !== $scheme) {
+            throw new UnsupportedSchemeException($dsn, 'smsbiuras', $this->getSupportedSchemes());
+        }
+
+        $uid = $this->getUser($dsn);
+        $apiKey = $this->getPassword($dsn);
+        $from = $dsn->getRequiredOption('from');
+        $test = $dsn->getOption('test', false);
```

Comment:
When the option is provided by DSN, you'll get a string.
You should explicitly convert it to boolean with something like 
`filter_var($dsn->getOption('test', false), \FILTER_VALIDATE_BOOLEAN)`

Reply:
I am not quite happy how we activate the test mode via DSN. Any other idea @jderusse ?

Anyway I would propose to rename it to "test_mode"

Any maybe use "on" and "off" for toggeling instead of magic/Boolean numbers?